### ISSUE TITLE
Document the order of the coefficients returned by KrigingResult.getTrendCoefficients

### ISSUE THREAD

**Issue by mdelozzo** [openturns/openturns#2365]
Since openturns 1.21, `KrigingResult.getTrendCoefficients` no longer returns a collection of `Point` but a unique `Point`.

First, a minor point: [in the documentation](https://openturns.github.io/openturns/latest/user_manual/response_surface/_generated/openturns.KrigingResult.html#openturns.KrigingResult.getTrendCoefficients),  I think that `Point` would be more relevant than _sequence of float_ (in 1.20, we can read _collection of `Point`_).

Then, a major point: surely there is a good explanation for this API change and so I'm not criticizing that. But I would like to share a feeling. It is now more difficult to read the coefficient associated with a given basis function. To illustrate that, I built a Kriging from some samples of $f(x)=(1+2x,3+4x)$ and I got `[1,3,2,4,1.60118e-15,-1.37244e-15]` while I would have expected `[1,2,1.60118e-15,3,4,1.60118e-15]`. In other words, I would have expected: first the coefficients of the first output, then those of the second output. 

I think that documenting the order of the coefficients would be a good thing. Not only in this case of 1 input and $p>1$ outputs but also in the general case of $d$ inputs and $p$ outputs.

**Comment by sofianehaddad** [comment]
I will update the documentation
you might notice we grouped coefficients by blocks ( (1,3), (2,4), (1.60118e-15,-1.37244e-15)).

 In other words, $\phi_0$,..., $\phi_{m-1}$, `m` being the size of the basis, $\phi_j$ a p-vectorial function

**Comment by mdelozzo** [comment]
Something like $(\phi_{1,1}\ldots\phi_{1,p}\ldots\phi_{m,1}\ldots\phi_{m,p})$ would help.

**Comment by mbaudin47** [comment]
Indeed, the notations in this page are confusing, wrong and inconsistent with other help pages: it would be too long to point them all here.
Notice that "sequence of float" might be the right naming. Indeed, if you give to the class a list of float, the conversion to a Point is done at the Swig layer. So anything like a tuple of floats or even a 1d-numpy array will do the job. We cannot list them all everywhere, hence the name "sequence of floats".

**Comment by mdelozzo** [comment]
I agree that _sequence of floats_ is the right naming for the type of the argument `trendCoefficients` passed to the constructor of `KrigingResult` (note that the documentation only indices _`Point`_) but it doesn't seem to be the right one for the type of the object returned by `KrigingResult`, as you can see with this reproducer:

```python
import openturns as ot

f = ot.SymbolicFunction(["x"], ["x * sin(x)"])
sampleX = [[1.0], [2.0], [3.0], [4.0], [5.0], [6.0]]
sampleY = f(sampleX)
basis = ot.Basis(
    [ot.SymbolicFunction(["x"], ["x"]), ot.SymbolicFunction(["x"], ["x^2"])]
)
covarianceModel = ot.GeneralizedExponential([2.0], 2.0)
algoKriging = ot.KrigingAlgorithm(sampleX, sampleY, covarianceModel, basis)
algoKriging.run()

resKriging = algoKriging.getResult()
print(type(resKriging.getTrendCoefficients()))  # Point

result = ot.KrigingResult(
    sampleX,
    sampleY,
    resKriging.getMetaModel(),
    resKriging.getResiduals(),
    resKriging.getRelativeErrors(),
    basis,
    list(resKriging.getTrendCoefficients()),  # trendCoefficients as a list[float]
    covarianceModel,
    resKriging.getCovarianceCoefficients(),
)

print(type(result.getTrendCoefficients()))  # Point
```

**Comment by mbaudin47** [comment]
Ok I now see your specific point. The method `getTrendCoefficients()` really returns a `Point`. My text was right only for input arguments of OT's methods (where "sequence of floats" is OK), not output of a given method (where `Point` is the correct name).
